### PR TITLE
fix static check errors in test/integration/etcd

### PR DESCRIPTION
**Pull request by 928234269**
Signed-off-by: Sakura <longfei.shang@daocloud.io>

**What type of PR is this?**
> /kind cleanup

**What this PR does / why we need it**:
cleanup staticcheck errors, it says as below:
```
test/integration/etcd/server.go:128:2: the goroutine calls T.Fatal, which must be called in the same goroutine as the test (SA2002)
test/integration/etcd/server.go:177:26: kubeClient.Discovery().ServerResources is deprecated: use ServerGroupsAndResources instead.  (SA1019)
```

**Which issue(s) this PR fixes**:
Ref:#81657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/integration/etcd/server.go [hunk 1]
```diff
@@ -125,23 +125,21 @@ func StartRealMasterOrDie(t *testing.T, configFuncs ...func(*options.ServerRunOp
...
+		if err := recover(); err != nil {
+			t.Errorf("Unexpected panic trying to start API master: %#v", err)
 		}
 	}()
 
+	prepared, err := kubeAPIServer.PrepareRun()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := prepared.Run(stopCh); err != nil {
```
Comment (liggitt) [inline test/integration/etcd/server.go]: this is not equivalent. Run() blocks, so moving this outside the goroutine will prevent the test from running/completing

Comment (928234269) [reply test/integration/etcd/server.go]: So i need to keep goroutine When fixing this error?

Comment (928234269) [reply test/integration/etcd/server.go]: I made a modified submission please review it. thinks!

**Comment by liggitt** [comment]
lgtm, please squash to a single commit

**Comment by 928234269** [comment]
@liggitt I had some issues with my local git that prevented me from merging commits, so I opened a Pr with a new branch. I will close this pr. The new Pr is here.#87457.

**Pull request closed without merging**